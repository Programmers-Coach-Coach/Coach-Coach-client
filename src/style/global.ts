import { createGlobalStyle } from "styled-components";

export const GlobalStyle = createGlobalStyle`
    @font-face {
        font-family: 'Pretendard';
        font-style: normal;
        font-weight: 400;
        src: url('https://cdn.jsdelivr.net/npm/pretendard@1.3.5/dist/web/static/pretendard.css');
    }
    body {
        padding : 0;
        width: 100%;
        max-width: 600px;
        margin: 0 auto;
        background-color: ${({ theme }) => theme.color.background};
        font-family: 'Pretendard', sans-serif;
    }
`;
