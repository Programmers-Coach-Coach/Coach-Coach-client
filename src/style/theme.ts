export type ColorKey =
  | "background"
  | "box"
  | "border"
  | "primary"
  | "secondary"
  | "third"
  | "error"
  | "likes"
  | "review"
  | "text"
  | "gray1"
  | "gray2"
  | "gray3"
  | "lightTransparentBlack";

export type FontSize = "large" | "medium" | "small";
export type FontWeight = "bold" | "normal";
export type ButtonSize =
  | "large"
  | "small"
  | "full"
  | "mini"
  | "super-mini"
  | "full-sharp";
export type CustomButtonType = "contained" | "outlined";
export type ProfileImageSize = "mini" | "small" | "medium" | "large";
export type ModalSize = "default";
export type PaddingSize = "default";

export type TitleSize = "t1" | "t2";
export type BodySize = "b1" | "b2" | "b3";

export type Sports = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12;

interface Theme {
  color: Record<ColorKey, string>;
  fontSize: Record<FontSize, string>;
  fontWeight: Record<FontWeight, string>;
  button: {
    [key in ButtonSize]: {
      padding: string;
      height: string;
      width?: string;
    };
  };
  borderRadius: {
    default: string;
  };
  boxShadow: string;
  profileImage: {
    [key in ProfileImageSize]: {
      width: string;
      height: string;
    };
  };
  modal: {
    [key in ModalSize]: {
      width: string;
    };
  };
  padding: {
    [key in PaddingSize]: string;
  };
  titleSize: {
    [key in TitleSize]: {
      fontSize: string;
      lineHeight: string;
      bold: string;
    };
  };
  bodySize: {
    [key in BodySize]: {
      fontSize: string;
      lineHeight: string;
      bold: string;
    };
  };
  buttonVariant: {
    [key in CustomButtonType]: {
      backgroundColor: string;
      color: string;
      border: string;
      borderHoverColor: string;
    };
  };
  sports: {
    [key in Sports]: string;
  };
}

export const theme: Theme = {
  color: {
    background: "#FFFFFF",
    box: "#F8F9FA",
    border: "#BDBDBD",
    primary: "#5271FF",
    secondary: "#9CABEF",
    third: "#AECDFF",
    error: "#F22455",
    likes: "#ABDEE6",
    review: "#FEAF29",
    text: "#212121",
    gray1: "#E1E1E1",
    gray2: "#D9D9D9",
    gray3: "#808080",
    lightTransparentBlack: "rgba(0, 0, 0, 0.3)"
  },
  fontSize: {
    large: "20px",
    medium: "16px",
    small: "12px"
  },
  fontWeight: {
    bold: "700",
    normal: "400"
  },
  button: {
    large: {
      padding: "16px",
      height: "48px",
      width: "314px"
    },
    small: {
      padding: "16px",
      height: "32px",
      width: "105px"
    },
    full: {
      padding: "16px",
      height: "48px",
      width: "100%"
    },
    mini: {
      padding: "16px",
      height: "32px",
      width: "86px"
    },
    "super-mini": {
      padding: "3px",
      height: "auto",
      width: "auto"
    },
    "full-sharp": {
      padding: "4px",
      height: "auto",
      width: "auto"
    }
  },
  borderRadius: {
    default: "20px"
  },
  boxShadow: "0 10px 40px 0 rgba(0, 0, 0, 0.1)",
  profileImage: {
    mini: {
      width: "60px",
      height: "60px"
    },
    small: {
      width: "86px",
      height: "86px"
    },
    medium: {
      width: "114px",
      height: "114px"
    },
    large: {
      width: "316px",
      height: "140px"
    }
  },
  modal: {
    default: {
      width: "352px"
    }
  },
  padding: {
    default: "50px"
  },
  titleSize: {
    t1: {
      fontSize: "20px",
      lineHeight: "36px",
      bold: "800"
    },
    t2: {
      fontSize: "16px",
      lineHeight: "24px",
      bold: "800"
    }
  },
  bodySize: {
    b1: {
      fontSize: "16px",
      lineHeight: "16px",
      bold: "400"
    },
    b2: {
      fontSize: "12px",
      lineHeight: "16px",
      bold: "400"
    },
    b3: {
      fontSize: "16px",
      lineHeight: "32px",
      bold: "800"
    }
  },
  buttonVariant: {
    contained: {
      backgroundColor: "#5271FF",
      color: "#FFFFFF",
      border: "none",
      borderHoverColor: "#5271FF"
    },
    outlined: {
      backgroundColor: "transparent",
      color: "#5271FF",
      border: "1px solid #5271FF",
      borderHoverColor: "#5271FF"
    }
  },
  sports: {
    1: "#F0C40E",
    2: "#B6D9DD",
    3: "#8369C2",
    4: "#4D8AA6",
    5: "#D1AA8D",
    6: "#F85F51",
    7: "#AAC77A",
    8: "#FFC1C2",
    9: "#2E4B77",
    10: "#B2CBE7",
    11: "#0496D1",
    12: "#5A4097"
  }
};
