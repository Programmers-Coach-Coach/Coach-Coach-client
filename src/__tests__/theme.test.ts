import "jest-styled-components";
import { theme, BodySize, TitleSize } from "../style/theme";
import {
  ColorKey,
  FontSize,
  FontWeight,
  ButtonSize,
  ProfileImageSize
} from "../style/theme";
describe("Theme Object", () => {
  it("should match the theme snapshot", () => {
    expect(theme).toMatchSnapshot();
  });

  test.each([
    ["background", "#234257"],
    ["yellow", "#FEAF29"],
    ["black", "#000000"],
    ["white", "#F8F9FA"],
    ["gray1", "#E1E1E1"],
    ["gray2", "#D9D9D9"],
    ["gray3", "#808080"],
    ["semiTransparentBlack", "rgba(0, 0, 0, 0.8)"],
    ["lightTransparentBlack", "rgba(0, 0, 0, 0.3)"]
  ] as [ColorKey, string][])(
    "should have correct color value for %s",
    (colorKey, expectedValue) => {
      expect(theme.color[colorKey]).toBe(expectedValue);
    }
  );

  test.each([
    ["large", "20px"],
    ["medium", "16px"],
    ["small", "12px"]
  ] as [FontSize, string][])(
    "should have correct fontSize value for %s",
    (fontSize, expectedValue) => {
      expect(theme.fontSize[fontSize]).toBe(expectedValue);
    }
  );

  test.each([
    ["bold", "700"],
    ["normal", "400"]
  ] as [FontWeight, string][])(
    "should have correct fontWeight value for %s",
    (fontWeight, expectedValue) => {
      expect(theme.fontWeight[fontWeight]).toBe(expectedValue);
    }
  );

  test.each([
    ["large", { padding: "16px", height: "48px", width: "100%" }],
    ["small", { padding: "16px", height: "32px" }]
  ] as [ButtonSize, { padding: string; height: string; width?: string }][])(
    "should have correct button value for %s",
    (buttonSize, expectedValue) => {
      expect(theme.button[buttonSize]).toEqual(expectedValue);
    }
  );

  it("should have correct borderRadius value", () => {
    expect(theme.borderRadius.default).toBe("20px");
  });

  it("should have correct boxShadow value", () => {
    expect(theme.boxShadow).toBe("0 10px 40px 0 rgba(0, 0, 0, 0.03)");
  });

  test.each([
    ["small", { width: "86px", height: "86px" }],
    ["medium", { width: "114px", height: "114px" }],
    ["large", { width: "316px", height: "140px" }]
  ] as [ProfileImageSize, { width: string; height: string }][])(
    "should have correct profileImage value for %s",
    (profileImageSize, expectedValue) => {
      expect(theme.profileImage[profileImageSize]).toEqual(expectedValue);
    }
  );
  it("should have correct modal value", () => {
    expect(theme.modal.default.width).toBe("352px");
  });

  it("should have correct padding value", () => {
    expect(theme.padding.default).toBe("50px");
  });

  test.each([
    ["t1", { fontSize: "20px", lineHeight: "36px", bold: "800" }],
    ["t2", { fontSize: "16px", lineHeight: "24px", bold: "800" }]
  ] as [TitleSize, { fontSize: string; lineHeight: string; bold: string }][])(
    "should have correct titleSize value for %s",
    (titleSize, expectedValue) => {
      expect(theme.titleSize[titleSize]).toEqual(expectedValue);
    }
  );

  test.each([
    ["b1", { fontSize: "16px", lineHeight: "16px", bold: "400" }],
    ["b2", { fontSize: "12px", lineHeight: "16px", bold: "400" }],
    ["b3", { fontSize: "16px", lineHeight: "32px", bold: "800" }]
  ] as [BodySize, { fontSize: string; lineHeight: string; bold: string }][])(
    "should have correct titleSize value for %s",
    (bodySize, expectedValue) => {
      expect(theme.bodySize[bodySize]).toEqual(expectedValue);
    }
  );
});
