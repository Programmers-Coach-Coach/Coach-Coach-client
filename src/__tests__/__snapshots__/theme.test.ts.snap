// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme Object should match the theme snapshot 1`] = `
{
  "bodySize": {
    "b1": {
      "bold": "400",
      "fontSize": "16px",
      "lineHeight": "16px",
    },
    "b2": {
      "bold": "400",
      "fontSize": "12px",
      "lineHeight": "16px",
    },
    "b3": {
      "bold": "800",
      "fontSize": "16px",
      "lineHeight": "32px",
    },
  },
  "borderRadius": {
    "default": "20px",
  },
  "boxShadow": "0 10px 40px 0 rgba(0, 0, 0, 0.03)",
  "button": {
    "large": {
      "height": "48px",
      "padding": "16px",
      "width": "314px",
    },
    "small": {
      "height": "32px",
      "padding": "16px",
      "width": "105px",
    },
  },
  "buttonVariant": {
    "contained": {
      "backgroundColor": "#5271FF",
      "border": "none",
      "borderHoverColor": "#5271FF",
      "color": "#FFFFFF",
    },
    "outlined": {
      "backgroundColor": "transparent",
      "border": "1px solid #5271FF",
      "borderHoverColor": "#5271FF",
      "color": "#5271FF",
    },
  },
  "color": {
    "background": "#FFFFFF",
    "box": "#F8F9FA",
    "error": "#F22455",
    "gray1": "#E1E1E1",
    "gray2": "#D9D9D9",
    "gray3": "#808080",
    "lightTransparentBlack": "rgba(0, 0, 0, 0.3)",
    "likes": "#ABDEE6",
    "primary": "#5271FF",
    "review": "#FEAF29",
    "secondary": "#9CABEF",
    "text": "#212121",
    "third": "#AECDFF",
  },
  "fontSize": {
    "large": "20px",
    "medium": "16px",
    "small": "12px",
  },
  "fontWeight": {
    "bold": "700",
    "normal": "400",
  },
  "modal": {
    "default": {
      "width": "352px",
    },
  },
  "padding": {
    "default": "50px",
  },
  "profileImage": {
    "large": {
      "height": "140px",
      "width": "316px",
    },
    "medium": {
      "height": "114px",
      "width": "114px",
    },
    "small": {
      "height": "86px",
      "width": "86px",
    },
  },
  "sports": {
    "0": "#F0C40E",
    "1": "#B6D9DD",
    "10": "#0496D1",
    "11": "#5A4097",
    "2": "#8369C2",
    "3": "#4D8AA6",
    "4": "#D1AA8D",
    "5": "#F85F51",
    "6": "#AAC77A",
    "7": "#FFC1C2",
    "8": "#2E4B77",
    "9": "#B2CBE7",
  },
  "titleSize": {
    "t1": {
      "bold": "800",
      "fontSize": "20px",
      "lineHeight": "36px",
    },
    "t2": {
      "bold": "800",
      "fontSize": "16px",
      "lineHeight": "24px",
    },
  },
}
`;
